core/ts/__tests__/MaciState.test.ts

```diff
@@ -240,50 +226,26 @@ describe("MaciState", function () {
         maciState.polls[pollId].publishMessage(message, ecdhKeypair.pubKey);
       }
 
-      // processMessages() should fail if the state and message AQs are
-      // not merged yet
-      expect(() => {
-        maciState.polls[pollId].processMessages();
-      }).to.throw;
-
-      // Merge the state aq
-      maciState.stateAq.mergeSubRoots(0);
-      maciState.stateAq.merge(STATE_TREE_DEPTH);
-
-      expect(() => {
-        maciState.polls[pollId].processMessages();
-      }).to.throw;
-
-      // Merge the message aq
-      maciState.polls[pollId].messageAq.mergeSubRoots(0);
-      maciState.polls[pollId].messageAq.merge(treeDepths.messageTreeDepth);
-
-      expect(maciState.polls[pollId].messageAq.numLeaves).to.eq(messageBatchSize * 2);
+      // 48 messages in total
+      expect(maciState.polls[pollId].messages.length).to.eq(48);
 
       expect(maciState.polls[pollId].currentMessageBatchIndex).to.eq(undefined);
       expect(maciState.polls[pollId].numBatchesProcessed).to.eq(0);
 
       // Process messages
-      maciState.polls[pollId].processMessages();
+      maciState.polls[pollId].processMessages(pollId);
 
       // currentMessageBatchIndex is 0 because the current batch starts
       // with index 0.
       expect(maciState.polls[pollId].currentMessageBatchIndex).to.eq(0);
       expect(maciState.polls[pollId].numBatchesProcessed).to.eq(1);
 
       // Process messages
-      maciState.polls[pollId].processMessages();
+      maciState.polls[pollId].processMessages(pollId);
 
       expect(maciState.polls[pollId].currentMessageBatchIndex).to.eq(0);
       expect(maciState.polls[pollId].numBatchesProcessed).to.eq(2);
 
-      // Attempt to process messages, but this should fail as there are
-      // no more messages to process
-      // TODO: use VError to test for specific errors
-      expect(() => {
```

Comment:
I removed following assertions from `Poll.processMessages()` so it will never throw:

```
assert(this.isMessageAqMerged());
assert(this.messageAq.hasRoot(this.treeDepths.messageTreeDepth));
```

Reply:
fair enough